superset/daos/tag.py

```diff
@@ -257,3 +266,90 @@ def get_tagged_objects_for_tags(
                 for obj in saved_queries
             )
         return results
+
+    @staticmethod
+    def user_favorite_tag(tag_id: int) -> None:
+        """
+        Marks a specific tag as a favorite for the current user.
+        This function will find the tag by the provided id,
+        create a new UserFavoriteTag object that represents
+        the user's preference, add that object to the database
+        session, and commit the session. It uses the currently
+        authenticated user from the global 'g' object.
+        Args:
+            tag_id: The id of the tag that is to be marked as
+                    favorite.
+        Raises:
+            Any exceptions raised by the find_by_id function,
+            the UserFavoriteTag constructor, or the database session's
+            add and commit methods will propagate up to the caller.
+        Returns:
+            None.
+        """
+        tag = TagDAO.find_by_id(tag_id)
+        user = g.user
+
+        if tag and user:
+            tag.users_favorited.append(user)
+            db.session.add(tag)
```

Comment:
should we use session.merge as this tag object is returned object from the DAO?

Reply:
whats the difference between just doing add/commit?

Reply:
see https://docs.sqlalchemy.org/en/20/orm/session_state_management.html
and https://stackoverflow.com/questions/1849567/can-sqlalchemys-session-merge-update-its-result-with-newer-data-from-the-data

I believe since you get tag from the TagDAO the tag object is already in the session.
Calling add here is redundant.
But I am not sure about that `g.user` if that's returning a user object in the same session or if it's in a different session.
If it's in different sqla session, you will want to call merge() here so that it merges the user object to current session by matching the user.id.

Regardless, I think merge() is safer.

Reply:
Agree. We should pass the user as a param as opposed to using the global